### DISCUSSION TITLE
Postgres wire protocol for the cache

### DISCUSSION THREAD

**Discussion by abcpro1** [getdozer/dozer#2183] [Feature Requests]
Many databases implement the Postgres wire protocol for their SQL interface. Postgres wire protocol automatically gives access to JDBC, ODBC, the `psql` command line client, in addition to all other existing Postgres clients. 

If we put a SQL interface on the cache, what SQL features will we support? Currently, in gRPC and REST we have `limit`, `offset` and `filter`.

Postgres wire protocol also supports streaming. Will we also support this?

**Comment by chubei** [comment]
We just had a call and the conclusion is that implementing the sql engine using datafusion is the way to go.

For datafusion, we just have to see if we can implement the [`TableProvider`](https://docs.rs/datafusion/latest/datafusion/datasource/provider/trait.TableProvider.html) trait to use cache as a table.

And then we support whatever sql features that datafusion gives us.

**Comment by abcpro1** [comment]
Datafusion has no JSON support (no JSON data type and no JSON functions).
@chubei @v3g42

**Reply by chubei** [reply]
Yes, and we convert json to string when exporting to arrow. Let's skip that then.
